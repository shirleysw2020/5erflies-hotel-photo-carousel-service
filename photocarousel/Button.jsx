import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { ThemeProvider } from 'styled-components';

// refactored class component to use hooks
const Button = ({list, openModal, show}) => {
  return (
    <PhotoButton onClick={openModal}>Show all photos</PhotoButton>
  )
}

const PhotoButton = styled.button`
  position: absolute;
  bottom: 25px;
  right: 30px;
  width: 146px;
  cursor: pointer;
  letter-spacing: 0.2px;
  font-size: 14px;
  line-height: 18px;
  font-weight: 450;
  border-radius: 7px;
  color: #000;
  height: 30px;
  background-color: #F7F7F7;
  font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, "Helvetica Neue", sans-serif;
  border-width: 1px;
`;

export default Button;