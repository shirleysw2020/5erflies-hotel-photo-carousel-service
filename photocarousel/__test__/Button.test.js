import React from "react";
import { create } from "react-test-renderer";
import { shallow, mount } from 'enzyme';
import Button from '../Button.jsx';

describe("Button component", () => {
  test("Matches the snapshot", () => {
    const button = create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  // it ('debugging Button', () => {
  // const shallowWrapper = shallow(<Button />);
  // const mountWrapper = mount(<Button />);
  // console.log('shallow Button', shallowWrapper.debug());
  // console.log('mount Button', mountWrapper.debug());
  // })
});



// const openModal = jest.fn();
// describe('Button component', () => {
//   it('button click should invoke function openModal', () => {
//     const component = shallow(<Button onClick={openModal}/>);
//     component
//       // .find expects a node
//       .find('sc-bwzfXH kVobZv')
//       .simulate('click');
//     expect(openModal).toHaveBeenCalled();
//   });
// });

