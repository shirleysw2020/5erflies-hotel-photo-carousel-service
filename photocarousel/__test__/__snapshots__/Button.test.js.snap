// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<button
  className="sc-bdVaJa gwQknd"
>
  Show all photos
</button>
`;
